### ISSUE TITLE
Question on training without attribute in 4.4. Ablation Study on Using Attributes

### ISSUE THREAD

**Issue by e96031413** [bighuang624/AGAM#3]
Hi, if I would like to train AGAM on CUB dataset without attribute, how should I modify the source code to achieve the result shown in Table 4.?

**Comment by bighuang624** [comment]
@e96031413  We removed the attributes-guided branch, so the model in `model.py` was implemented like this

```py
class ProtoNetAGAMwoAttr(nn.Module):
    def __init__(self, backbone, semantic_size, out_channels):
        super(ProtoNetAGAMwoAttr, self).__init__()
        self.encoder = get_backbone(backbone)

        self.ca_block = CABlock(out_channels)
        self.sa_block = SABlock()

    def forward(self, inputs):

        embeddings = self.encoder(inputs.view(-1, *inputs.shape[2:]))
        ca_embeddings, ca_weights = self.ca_block(embeddings)
        embeddings = ca_embeddings
        sa_embeddings, sa_weights = self.sa_block(embeddings)
        embeddings = sa_embeddings

        return embeddings.view(*inputs.shape[:2], -1)
```

**Comment by e96031413** [comment]
Hello, since there are no sca_weights and ssa_weights in your provided code, should I keep the get_addition_loss function in utils.py? or how to modify it? Thanks

**Comment by bighuang624** [comment]
@e96031413  Sure, no need to keep the addition loss. Just calculate a prototypical loss with the prototypes and query features that are both refined by `ca_block` and `sa_block`. In other words: 

```py
train_loss = prototypical_loss(prototypes, query_embeddings, query_targets)
```

**Comment by e96031413** [comment]
Thanks, the code works successfully.
